dev_docs/key_concepts/anatomy_of_a_plugin.mdx

```diff
@@ -39,27 +39,41 @@ plugins/
 
 ```
 {
-  "id": "demo",
-  "version": "kibana",
+  "id": "examplePluginId",
+  "version": "1.0.0",
+  "kibanaVersion": "7.14.0",
   "server": true,
   "ui": true,
-  "owner": { [1]
+  "owner": {
     "name": "App Services",
     "githubTeam": "kibana-app-services"
   },
-  "description": "This plugin extends Kibana by doing xyz, and allows other plugins to extend Kibana by offering abc functionality. It also exposes some helper utilities that do efg", [2]
-  "requiredPlugins": ["data"], [3]
-  "optionalPlugins": ["alerting"] [4]
-  "requiredBundles": ["anotherPlugin"] [5]
+  "description": "A description about this plugin!",
+  "requiredPlugins": ["data"],
+  "optionalPlugins": ["alerting"]
+  "requiredBundles": ["anotherPlugin"]
 }
 ```
 
-[1], [2]: Every internal plugin should fill in the owner and description properties.
+`id` - [Required] The id of your plugin can be anything, though it should be fairly unique, as every plugin in an installation needs to be unique. It must be snakeCase.
+
+`version` - [Required] Note the version of your plugin. Kibana doesn't use this field, but it is required.
+
+`kibanaVersion` - [Optional] If your plugin is only compatible with a specific version of Kibana, put it here. Note that internal, first-party plugins shouldn't use this, as they are expected to be compatible with every Kibana version.
```

Comment:
> Note that internal, first-party plugins shouldn't use this, as they are expected to be compatible with every Kibana version.

This logic is not implemented yet. But we are going to remove this check https://github.com/elastic/kibana/issues/61087 

If `kibanaVersion` is not specified, the plugin system will use `version` field to check the compatibility with the current Kibana version. That's the reason tests are falling on your branch for plugins declaring `version < 8.0.0`.

Reply:
> we are going to remove this check #61087

I'm sorry, I read through that issue but I still don't understand exactly what the issue is now and what the desired behavior will be after the issue is fixed.   The issue makes "If kibanaVersion is not specified, the plugin system will use version field to check the compatibility with the current Kibana version." sound like that's how it's _supposed_ to work, but because of the bug, it's _not_ actually working that way? 

How would you define these two fields now (perhaps noting the issue and how it will work in the future)?

Reply:
Ah, never mind, I see now what you mean now that I looked at the failing tests.

I can't yet remove `kibanaVersion` from first-party plugins because of that fallback.  lmk what you think of the updated descriptions for these properties.